File: Python/code/T26/test_T26.py
import os
import Test1.test as file1
import Test2.test as file2
from Test1.test import N

files_to_compare = [
    ("Test1/test.py", "Test1_Nuitka/test.py"),
    ("Test2/test.py", "Test2_Nuitka/test.py")
]

def test_files_identical():
    for orig, nuitka in files_to_compare:
        assert os.path.exists(orig), f"Original file not found: {orig}"
        assert os.path.exists(nuitka), f"Nuitka file not found: {nuitka}"

        with open(orig, "r", encoding="utf-8") as f1:
            content1 = f1.read()
        with open(nuitka, "r", encoding="utf-8") as f2:
            content2 = f2.read()

        assert content1 == content2, f"Files differ: {orig} != {nuitka}"

def test_identical_output_check_all():
    array1 = [0] * N
    array2 = [0] * N

    copy11 = list(array1)
    copy12 = list(array2)
    copy21 = list(array1)
    copy22 = list(array2)

    file1.test(copy11, copy12, N)
    file2.test(copy21, copy22, N)

    if not (copy11 == copy21):
        raise AssertionError(f"Mismatch detected between versions!")

    if not (copy12 == copy22):
        raise AssertionError(f"Mismatch detected between versions!")

if __name__ == "__main__":
    test_files_identical()
    test_identical_output_check_all()

File: Java/code/T26/test_T26.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.Arrays;

public class test_T26 {

    public static void main(String[] args) {
        test_files_identical();
        test_identical_output_check_all();
    }

    public static void test_files_identical() {
        String[][] filesToCompare = {
            {"Test1/test.java", "Test1_Xint/test.java"},
            {"Test2/test.java", "Test2_Xint/test.java"}
        };

        for (String[] pair : filesToCompare) {
            try {
                Path pathOrig = Paths.get(pair[0]);
                Path pathXint = Paths.get(pair[1]);

                if (!Files.exists(pathOrig) || !Files.exists(pathXint)) {
                    System.err.println("[ERROR] File not found: " + pair[0] + " or " + pair[1]);
                    System.exit(1);
                }

                java.util.List<String> f1Lines = Files.readAllLines(pathOrig);
                java.util.List<String> f2Lines = Files.readAllLines(pathXint);

                f1Lines.removeIf(line -> ignoreLine(line));
                f2Lines.removeIf(line -> ignoreLine(line));

                if (!f1Lines.equals(f2Lines)) {
                    System.err.println("[ERROR] Files differ (ignoring package & REPS): "
                            + pair[0] + " != " + pair[1]);
                    System.exit(1);
                }

            } catch (IOException e) {
                System.err.println("[ERROR] Error reading files: " + e.getMessage());
                System.exit(1);
            }
        }
    }

    private static boolean ignoreLine(String line) {
        String l = line.trim();

        return l.startsWith("package ")
            || l.startsWith("public static final int REPS")
            || l.startsWith("public static final long REPS")
            || l.contains("for (int reps")
            || l.contains("for (long reps");
    }

    public static void test_identical_output_check_all() {
        try {
            int n = Test1.test.N;

            int[] array1 = new int[n];
            int[] array2 = new int[n];

            int[] copy11 = Arrays.copyOf(array1, n);
            int[] copy12 = Arrays.copyOf(array2, n);
            int[] copy21 = Arrays.copyOf(array1, n);
            int[] copy22 = Arrays.copyOf(array2, n);

            Test1.test.test(copy11, copy12, n);
            Test2.test.test(copy21, copy22, n);

            if (!Arrays.equals(copy11, copy21)) {
                throw new AssertionError("Mismatch detected between versions!");
            }

            if (!Arrays.equals(copy12, copy22)) {
                throw new AssertionError("Mismatch detected between versions!");
            }

        } catch (Throwable e) {
            System.err.println("[ERROR] Execution error: " + e.getMessage());
            e.printStackTrace();
            System.exit(1);
        }
    }
}